torch/distributed/distributed_c10d.py

```
@@ -73,7 +76,25 @@ def __new__(cls, name):
                              "Gloo or MPI backend for collective operations "
                              "on CPU tensors.")
         elif value == Backend.UNDEFINED:
-            raise ValueError("Invalid backend: '{}'".format(name))
+            try:
+                if name not in sys.modules:
+                    # using "torch_" prefix to avoid module naming conflict/confusion 
+                    other_backend = importlib.import_module("torch_{}".format(name.lower()))
+                else:
+                    other_backend = sys.modules["torch_{}".format(name.lower())]
+
+                if other_backend.__dict__.get("name"):
+                    pg_name = other_backend.name()
+                elif other_backend.__dict__.get("ProcessGroup{}".format(name.lower().capitalize())):
+                    pg_name = "ProcessGroup{}".format(name.lower().capitalize())
+                else:
+                    raise NameError("Not found '{}' symbol in '{}' python module".format(pg_name, "torch_{}".format(name.lower())))
```

This error will be hidden by the `except NameError` below.

---

`pg_name` is undefined when you raise this error.